tests/test_bot.py

```diff
@@ -0,0 +1,263 @@
+        """
+        # Create the bot
+        bot = Bot()
+
+        # Mock the slack client
+        with open(MEMBERS, 'r') as f:
+            members = json.load(f)
+        bot.slack.api_call = mock.MagicMock(return_value=members)
+
+        # Test the botid
+        self.assertEqual(bot.botid, 'UTEST3210')
+        bot.slack.api_call.assert_called_once_with('users.list')
+
+        # Test memoization (second call does not go to API)
+        self.assertEqual(bot.botid, 'UTEST3210')
+        bot.slack.api_call.assert_called_once_with('users.list')
+
+        # Test atbotid filter
+        table = (
+            "hello <@UTEST3210>!",
+            "<@UTEST3210> weather now",
+            "weather now <@UTEST3210>",
+        )
+
+        for msg in table:
+            self.assertIsNotNone(bot.atbotid.search(msg))
+
+    def test_chat_handling_and_responses(self):
+        """
+        Tests the chat handling and responses for a stream.
+        """
+
+        # Load the messages and weather
+        with open(RTM_MSG, 'r') as f:
+            messages = [
+                json.loads(line.strip())
+                for line in f
+            ]
+
+        with open(WEATHER, 'r') as f:
+            weather = json.load(f)
+
+        # Create a bot and patch the slack client
+        bot = Bot()
+        bot._botid = "UTEST3210"
+        bot.darksky.forecast = mock.MagicMock(return_value=weather)
+        bot.slack.rtm_connect = mock.MagicMock(return_value=True)
+        bot.slack.rtm_read = mock.MagicMock(return_value=messages)
+        bot.slack.api_call = mock.MagicMock()
+
+        # Handle the messages
+        bot.read_rtm_channel()
+
+        # Default bot.post call args and kwargs
+        args = ('chat.postMessage',)
+        kwargs = {'as_user': True, 'channel': u'CTESTCHAN'}
+
+        # Closure to create kwargs for testing bot.post
+        def make_call(response):
+            kws = {'text': response}
+            kws.update(kwargs)
+            return mock.call(*args, **kws)
+
+        # Make assertions about the handling
+        self.assertEqual(len(bot.slack.api_call.mock_calls), 7)
+        bot.slack.api_call.assert_has_calls([
+            make_call(u"Currently in 20001 it is 54.4\xb0F and mostly cloudy. It will be mostly cloudy throughout the day."),
+            make_call(u"Light rain on Sunday and Monday, with temperatures rising to 62\xb0F on Wednesday."),
+            make_call("I have made 0 Dark Sky API calls today."),
+            make_call(u"Currently in 58054 it is 54.4\xb0F and mostly cloudy. It will be mostly cloudy throughout the day."),
+            make_call(u"Currently in 90210 it is 54.4\xb0F and mostly cloudy. It will be mostly cloudy throughout the day."),
+            make_call(u"Currently in 20742 it is 54.4\xb0F and mostly cloudy. It will be mostly cloudy throughout the day."),
+            make_call("I'm happy to chat about the weather,  unfortunately I don't understand what you're asking."),
```

Comment:
nit: i believe this is the only test case we have that insures that we don't allow user input to call arbitrary methods on the bot class (assuming some poor intern later added a `handle_secret_internal_only_message` method without reading the whole class). it should probably have a more explicit test case